src/robot/libraries/Collections.py

```diff
@@ -386,7 +389,7 @@ def lists_should_be_equal(self, list1, list2, msg=None, values=True,
         default = 'Lengths are different: %d != %d' % (len1, len2)
         _verify_condition(len1 == len2, default, msg, values)
         names = self._get_list_index_name_mapping(names, len1)
-        diffs = list(self._yield_list_diffs(list1, list2, names))
+        diffs = list(self._yield_list_diffs(list1, list2, names,ignore_order))
```

Comment:
This change, nor related tests, have nothing to do with title casing. Are you possibly using the same branch when implementing two different features? That won't work with pull requests, you need to have a separate branch for each PR.